torch/distributed/_tensor/random.py

```diff
@@ -11,69 +12,29 @@
-    """Sets the random number generator state of the specified device mesh.
+_rng_tracker: Optional["CudaRNGStateTracker"] = None
 
-    Args:
-        new_state (:class:`torch.ByteTensor`): The desired state.
-        device_mesh (:class:`DeviceMesh`): The device mesh to set the RNG state.
-
-    Returns:
-        None
-
-    .. warning::
-        Current implementation only supports a GPU device mesh.
-        If ``device_mesh`` is a sub-mesh and the calling rank is not a part of it,
-        `set_rng_state` will not set its GPU device's generator state.
-    """
-    assert isinstance(
-        device_mesh, DeviceMesh
-    ), f"expect a DeviceMesh but {type(device_mesh)} was passed in."
-
-    if device_mesh.get_coordinate() is not None:
-        # the current rank is in mesh
-        device_handle = _get_device_handle(device_mesh.device_type)
-        if device_handle:
-            device_handle.set_rng_state(new_state)
-        else:
-            raise NotImplementedError(
-                f"DTensor randomness only supports cuda/cuda-like device type, but got {device_mesh.device_type}"
-            )
-
-
-def get_rng_state(device_mesh: DeviceMesh) -> Tensor:
-    """Returns the random number generator state of the calling rank as a
-    :class:`torch.ByteTensor` object.
-
-    Args:
-        device_mesh (:class:`DeviceMesh`): The device mesh to return the RNG state of.
 
-    Returns:
-        A :class:`torch.ByteTensor` object that contains the random number generator state.
-
-    .. warning::
-        Current implementation only supports a GPU device mesh.
-        If ``device_mesh`` is a sub-mesh and the calling rank is not a part of it,
-        `get_rng_state` still returns its GPU device's generator state.
-    """
-    assert isinstance(
-        device_mesh, DeviceMesh
-    ), f"expect a DeviceMesh but {type(device_mesh)} was passed in."
+def is_rng_supported_mesh(device_mesh: DeviceMesh) -> bool:
+    # currently we only support correct RNG on cuda/cuda-like device
     device_handle = _get_device_handle(device_mesh.device_type)
-    if device_handle:
-        return device_handle.get_rng_state()
+    if device_handle and hasattr(device_handle, "set_rng_state"):
+        return True
     else:
-        raise NotImplementedError(
-            f"DTensor randomness only supports cuda/cuda-like device type, but got {device_mesh.device_type}"
+        warnings.warn(
+            f"DTensor random operators may not have complete support on {device_mesh.device_type} device mesh"
         )
+        return False
 
 
-def manual_seed(seed: int, device_mesh: DeviceMesh) -> None:
+def manual_seed(seed: int, device_mesh: DeviceMesh, tp_dim: int = 0) -> None:
     """Sets the seed for generating random numbers for the calling rank.
 
     Args:
         seed (int): The desired seed.
         device_mesh (:class:`DeviceMesh`): The device mesh to set the seed.
+        tp_dim (int, optional): The mesh dimension where to apply Tensor Parallel
```

Comment:
I don't think this `tp_dim` arg make sense as we haven't use it anywhere inside this function.

I think we should make this arg be sth like `rng_tracker_type` instead, and expose this `manual_seed` API as the API when user want to manually initialize the RNGTracker, and user should have the flexibility to switch between OffsetBasedRNGTracker and the TPRNGTracker.

This might be good as a follow up PR after the TPRNGTracker PR, but I think we should remove this `tp_dim` arg in this PR before merging